### PR TITLE
RSPEC-5732 Disabling content security policy frame-ancestors directive is security-sensitive

### LINKED ISSUE #1952: RSPEC-5732 Disabling content security policy frame-ancestors directive is security-sensitive
Implements [RSPEC-5732](https://jira.sonarsource.com/browse/RSPEC-5732) MMF-1898

For both JS/TS

Should raise on the `csp()`/`helmet()` call (depending on the `helmet-csp`/`helmet` module used) if the `frameAncestors` directive (alias `"frame-ancestors"`) is not set or on the `frameAncestors` directive when set to `["none"]`:

```
var app = express()

  app.use(csp({  // Sensitive: frame-ancestors is not set
    directives: {
      defaultSrc: ["'self'", 'example.com', 'code.jquery.com'],
    }
  }));

  app.use(csp({
    directives: {
      defaultSrc: ["'self'", 'example.com', 'code.jquery.com'],
      "frame-ancestors": ["'none'"] // Sensitive: frame-ancestors is set to none
    } 
  }));
```

when `contentSecurityPolicy` middleware is disabled/set to `false` NO NEED TO RAISE, because [rule RSPEC-5728](https://github.com/SonarSource/SonarJS/issues/1949) will be already triggered:
```
app.use(
    helmet({
      contentSecurityPolicy: false, // Sensitive for 5728 only
    })
  );
```

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/express.js/RSPEC-5732

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #1952

### CODE REVIEW

File: eslint-bridge/tests/rules/frame-ancestors.test.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      },
+      {
+        code: `
+        const helmet = require('helmet');
+        const express = require('express');
+        const app = express();
+        app.use(
+          helmet.contentSecurityPolicy({
+            directive: {
+              "frame-ancestors": ['none']
```
Comment (andrey-tyukin-sonarsource) [inline eslint-bridge/tests/rules/frame-ancestors.test.ts:216]: Here also: `"'none'"` <-> `'none'`

File: eslint-bridge/src/rules/frame-ancestors.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import { Rule } from 'eslint';
+import * as estree from 'estree';
+import { getModuleNameOfNode, getObjectExpressionProperty } from './utils';
+import { Express } from './utils-express';
+
+const HELMET = 'helmet';
+const HELMET_CSP = 'helmet-csp';
+const DIRECTIVE = 'directive';
+const NONE = 'none';
```
Comment (andrey-tyukin-sonarsource) [inline eslint-bridge/src/rules/frame-ancestors.ts:30]: Shouldn't this be `'\'none\''` or `"'none'"`?

**Review by andrey-tyukin-sonarsource — CHANGES_REQUESTED** [review]
Overall structure seems alright, but a few string constants seem to require double-checking. Consider the following example from the spec:
```
app.use(csp({
    directives: {
      defaultSrc: ["'self'", 'example.com', 'code.jquery.com'],
      "frame-ancestors": ["'none'"] // Sensitive: frame-ancestors is set to none
    } 
  }));
```

  * It uses `directives`-with-`s` property
  * It uses `"'none'"` (double quotes for JS, single quotes for header fragments)

Same in the example from `helmet`-documentation itself:
```
app.use(
  contentSecurityPolicy({
    directives: {
      defaultSrc: ["'self'", "default.example"],
      scriptSrc: ["'self'", "'unsafe-inline'"],
      objectSrc: ["'none'"],
      upgradeInsecureRequests: [],
    },
    reportOnly: false,
  })
);
```
Note the `directives` and the `"'single-double-quotes'"`.

File: eslint-bridge/src/rules/frame-ancestors.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// https://jira.sonarsource.com/browse/RSPEC-5732
+
+import { Rule } from 'eslint';
+import * as estree from 'estree';
+import { getModuleNameOfNode, getObjectExpressionProperty } from './utils';
+import { Express } from './utils-express';
+
+const HELMET = 'helmet';
+const HELMET_CSP = 'helmet-csp';
+const DIRECTIVE = 'directive';
```
[the remark this answers is not in the document]
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/rules/frame-ancestors.ts]: My bad, it seems to be plural.

**Review by andrey-tyukin-sonarsource — APPROVED** [review]

**Pull request merged**